api/channel.go

```diff
@@ -843,17 +843,25 @@ func setLastViewedAt(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	data := model.StringInterfaceFromJson(r.Body)
 	newLastViewedAt := int64(data["last_viewed_at"].(float64))
+	channelName := data["channel_name"].(string)
 
 	Srv.Store.Channel().SetLastViewedAt(id, c.Session.UserId, newLastViewedAt)
 
-	preference := model.Preference{
+	chanPref := model.Preference{
 		UserId:   c.Session.UserId,
-		Category: model.PREFERENCE_CATEGORY_LAST,
+		Category: c.TeamId,
```

Comment:
~~Can you swap the Category and Name here so that the Category is 'last_channel' or whatever and the Name is the team id? That'll keep it consistent with how the team-specific themes are defined.~~

Also, have you thought about adding code to remove the old last channel preferences since they won't be used any more? It could be in a separate PR, but I think we might want to be careful about leaving unused preferences around since they'll increase the initial load slightly for every user